airbyte-ci/connectors/cdk/cdk/java_cdk/tasks.py

```diff
@@ -0,0 +1,109 @@
+"""CI Workflow tasks for Airbyte Java CDK"""
+
+import os
+
+import pygit2
+from aircmd.actions.environments import with_gradle
+from aircmd.models.base import PipelineContext
+from aircmd.models.utils import load_settings, make_pass_decorator
+from dagger import Client, Container
+from prefect import task
+from prefect.artifacts import create_link_artifact
+
+from .settings import JavaCDKSettings
+
+pass_pipeline_context = make_pass_decorator(PipelineContext, ensure=True)
+pass_global_settings = make_pass_decorator(JavaCDKSettings, ensure=True)
+
+
+@task
+async def build_java_cdk_task(client: Client, settings: JavaCDKSettings, ctx: PipelineContext, scan: bool = False) -> Container:
+    files_from_host = [
+        "./build.gradle",
+        "./gradlew",
+        "./gradle",
+        "./airbyte-cdk/java/**",
+        "./publish-repositories.gradle",
+        "./gradle.properties",
+        "./deps.toml",
+        "./build.gradle",
+        "./settings.gradle",
+        "./.env",
+        "./.env.dev",
+    ]
+
+    repo = pygit2.Repository(".")
+    repo_root = os.path.dirname(os.path.dirname(repo.path))
+    print(repo_root)
+    mount_dir = client.host().directory(repo_root, include=files_from_host)
+
+    gradle_command = ["./gradlew", ":airbyte-cdk:java:airbyte-cdk:assemble", "publishToMavenLocal", "--build-cache"]
+
+    result = (
+        with_gradle(client, ctx, settings, bind_to_docker_host=False)
+        .with_mounted_directory("/airbyte", mount_dir)
+        .with_(load_settings(settings))
+        .with_exec(gradle_command + ["--scan"] if scan else gradle_command)
+        .with_exec(["rsync", "-az", "/root/.gradle/", "/root/gradle-cache"])
+    )
+
+    if scan:
+        await create_link_artifact(
+            key="gradle-build-scan",
+            link=result.file("/airbyte/scan-journal.log").contents().split(" - ")[2].strip(),
+            description="Gradle build scan",
+        )
+
+    return result.sync()
```

Comment:
If you want `await build_java_cdk_task(**)` to actually run the `with_exec` I think you should await the container here.
Otherwise you'll get an unevaluated container that you'd have to await downstream.
If awaiting downstream is what you want you should not use `.sync` and directly return `result`. 
It will give the freedom to the caller to add additional steps on the container.
Also, container object are now directly awaitable: `await result` has the same effect as `await result.sync()`

Reply:
Yeah, I've waffled back and forth on this, but i think the tl;dr here is that the task file should not be the one that controls whether or not the container is actually executed, but only the flows/ci.py.

Reply:
I think I will add this refactor in as part of this pr

Reply:
Also, i think it's cool that containers are now directly awaitable, when was this introduced? I think this is much better ux in general. One thing that I think that is annoying about python is that there's no easy syntax for unwrapping nested futures, so you end up with a lot of verbose awaits. Making the container directly awaitable is much better.